SpatialGDK/Source/SpatialGDK/Private/EngineClasses/SpatialActorChannel.cpp

```diff
@@ -252,31 +253,60 @@ void USpatialActorChannel::DeleteEntityIfAuthoritative()
 	}
 
 	bool bHasAuthority = NetDriver->IsAuthoritativeDestructionAllowed() && NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialGDK::Position::ComponentId);
-
-	if (bHasAuthority)
+	if (Actor != nullptr)
 	{
-		if (Actor != nullptr)
+		bool bIsTearOff = Actor->GetTearOff();
+		bool bIsNetStartup = Actor->IsNetStartupActor();
+		DelayedRetireCallback FuncRetireActor = [bIsTearOff, bIsNetStartup](USpatialNetDriver* NetDriver, USpatialSender* Sender, Worker_EntityId EntityId)
 		{
 			// Workaround to delay the delete entity request if tearing off.
 			// Task to improve this: UNR-841
+			if (bIsTearOff)
+			{
+				// To ensure that the actor has time to stops replicating to the runtime.
+				NetDriver->DelayedRetireEntity(EntityId, 1.0f, bIsNetStartup);
+			}
+			else
+			{
+				Sender->RetireEntity(EntityId, bIsNetStartup);
+			}
+		};
+		if (bHasAuthority)
+		{
 			if (Actor->GetTearOff())
 			{
-				NetDriver->DelayedRetireEntity(EntityId, 1.0f, Actor->IsNetStartupActor());
 				// Since the entity deletion is delayed, this creates a situation,
 				// when the Actor is torn off, but still replicates.
 				// Disabling replication makes RPC calls impossible for this Actor.
 				Actor->SetReplicates(false);
 			}
-			else
+			FuncRetireActor(NetDriver, Sender, EntityId);
+		}
+		else
```

Comment:
We only want to do this if `!NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialGDK::Position::ComponentId)`
If `IsAuthoritativeDestructionAllowed` is false we don't want to delay deletion, we just ignore it.

Reply:
In fact I think we *only* want to do this if `NetDriver->StaticComponentView->HasAuthority(EntityId, SpatialGDK::Position::ComponentId)` is false, but local authority is true.